sips/fork_support.md

```diff
@@ -20,56 +20,209 @@ Client implementation should calculate when a new duty starts to decide if it st
-    JatoTestnet    = DomainType{0x0, 0x0, JatoChain, 0x1}
+	GenesisMainnet = DomainType{0x0, 0x0, byte(MainnetNetworkID), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, byte(PrimusNetworkID), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
+	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoV2NetworkID), 0x1}
 )
 
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
 type ForkData struct {
-    Epoch  phase0.Epoch
-    Domain DomainType
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType
+}
+
+func (domainType DomainType) GetNetworkID() NetworkID {
+	return NetworkID{domainType[2]}
+}
+
+func (networkID NetworkID) GetForksData() []*ForkData {
+	switch networkID {
+	case MainnetNetworkID:
+		return mainnetForks()
+	case PrimusNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: PrimusTestnet}}
+	case JatoNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoTestnet}}
+	case JatoV2NetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoV2Testnet}}
+	default:
+		return []*ForkData{}
+	}
+}
+
+// mainnetForks returns all forks for the mainnet chain
+func mainnetForks() []*ForkData {
+	return []*ForkData{
+		{
+			Epoch:  0,
+			Domain: GenesisMainnet,
+		},
+	}
 }
 
-func (domainType DomainType) GetChain() byte {
-    return domainType[2]
+func (networkID NetworkID) DefaultFork() *ForkData {
+	return networkID.GetForksData()[0]
 }
 
-func (domainType DomainType) GetForksData() []*ForkData {
-    switch domainType.GetChain() {
-    case GenesisChain:
-        return genesisForks()
-    default:
-        return []*ForkData{}
-    }
+// GetCurrentFork returns the ForkData with highest Epoch smaller or equal to "epoch"
+func (networkID NetworkID) ForkAtEpoch(epoch phase0.Epoch) (*ForkData, error) {
+	// Get list of forks
+	forks := networkID.GetForksData()
+
+	// If empty, raise error
+	if len(forks) == 0 {
+		return nil, errors.New("Fork list by GetForksData is empty. Unknown Network")
+	}
+
+	var current_fork *ForkData
+	for _, fork := range forks {
+		if fork.Epoch <= epoch {
+			current_fork = fork
+		}
+	}
+	return current_fork, nil
 }
 
-func genesisForks() []*ForkData {
-    return []*ForkData{
-        {
-            Epoch:  0,
-            Domain: GenesisMainnet,
-        },
-    }
+func (f ForkData) GetRoot() ([]byte, error) {
+	byts, err := json.Marshal(f)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal ForkData")
+	}
+	ret := sha256.Sum256(byts)
+	return ret[:], nil
 }
+```
+
+### Structures access to types
+
+- `Share`:
+  - It's more appropriate for the structure to hold the information on the network that it's in, instead of the domain type. Thus, _Share_ will contain a `NetworkID` field instead of a `DomainType` field.
```

Comment:
didn't we say we are reverting this change?

Reply:
If we don't do this change implementation team will be having issue imo

Reply:
@MatheusFranco99 
After talking to Alon we will keep DomainType in share in the SIP.

Since currently Share is held by the `Validator` it may seem problematic.

However since there are no tests for `Validator` we can either easily change it in spec code or even maybe remove it (if it isn't tested maybe no need for it).

This will allow engineering to be able to work with the SIP because they won't be constrained on where `share` is placed